src/VisualStudio/NuGet.Packaging.VisualStudio/IMsBuildService.cs

```diff
@@ -0,0 +1,10 @@
+﻿using Clide;
+using Microsoft.Build.Execution;
+
+namespace NuGet.Packaging.VisualStudio
+{
+	public interface IMsBuildService
+	{
+		BuildSubmission BeginBuild(string projectPath, string target = "Build");
```

Comment:
Maybe make this abstraction less leaky in case we need to totally change the implementation? Maybe something like:

```
interface IBuildService
{
    Task BuildAsync(string projectPath, string target = "Build");
}
```

And maybe the async-ness comes from doing `BuildSubmission.ExecuteAsync` where the callback sets the `TaskCompletionSource` as complete. Maybe it's all overkill and it should be a `void BeginBuild` for now since we're not waiting for it at all?